Marlin/ultralcd.cpp

```diff
@@ -1815,8 +1830,32 @@ void lcd_reset_alert_level() { lcd_status_message_level = 0; }
   void lcd_buttons_update() {
     #if ENABLED(NEWPANEL)
       uint8_t newbutton = 0;
-      if (READ(BTN_EN1) == 0) newbutton |= EN_A;
-      if (READ(BTN_EN2) == 0) newbutton |= EN_B;
+      #if BTN_EN1 > 0
+        if (READ(BTN_EN1) == 0) newbutton |= EN_A;
+      #endif
+      #if BTN_EN2 > 0
+        if (READ(BTN_EN2) == 0) newbutton |= EN_B;
+      #endif
+      #if ENABLED(RIGIDBOT_PANEL)
+        if (millis() > next_fake_encoder_update_ms && READ(BTN_UP) == 0) {
+          encoderDiff = -1 * ENCODER_STEPS_PER_MENU_ITEM;
+          next_fake_encoder_update_ms = millis() + 300;
+        }
+        if (millis() > next_fake_encoder_update_ms && READ(BTN_DWN) == 0) {
+          encoderDiff = ENCODER_STEPS_PER_MENU_ITEM;
+          next_fake_encoder_update_ms = millis() + 300;
+        }
+        if (millis() > next_fake_encoder_update_ms && READ(BTN_LFT) == 0) {
+          encoderDiff = -1 * ENCODER_PULSES_PER_STEP;
+          next_fake_encoder_update_ms = millis() + 300;
+        }
+        if (millis() > next_fake_encoder_update_ms && READ(BTN_RT) == 0) {
+          encoderDiff = ENCODER_PULSES_PER_STEP;
+          next_fake_encoder_update_ms = millis() + 300;
+        }
+      #endif
+
+
```

Comment:
```
  void lcd_buttons_update() {
     #if ENABLED(NEWPANEL)
       uint8_t newbutton = 0;
       if (READ(BTN_EN1) == 0) newbutton |= EN_A;
       #if BTN_EN1 > 0
         if (READ(BTN_EN1) == 0) newbutton |= EN_A;
       #endif
       #if BTN_EN2 > 0
         if (READ(BTN_EN2) == 0) newbutton |= EN_B;
       #endif
       #if ENABLED(RIGIDBOT_PANEL)
         millis_t now = millis();
         if (now > next_fake_encoder_update_ms) {
           if (READ(BTN_UP) == 0) {
             encoderDiff = -1 * ENCODER_STEPS_PER_MENU_ITEM;
           }
           if (READ(BTN_DWN) == 0) {
             encoderDiff = ENCODER_STEPS_PER_MENU_ITEM;
           }
           if (READ(BTN_LFT) == 0) {
             encoderDiff = -1 * ENCODER_PULSES_PER_STEP;
           }
           if (READ(BTN_RT) == 0) {
             encoderDiff = ENCODER_PULSES_PER_STEP;
           }
           next_fake_encoder_update_ms = now + 300;
         }
       #endif
```
is a bit less time and space consuming.

Calling mills() inside the interrupt is a bad idea.
Why is next_fake_encoder_update_ms used, why not the already existent next_button_update_ms?
Why is ENCODER_STEPS_PER_MENU_ITEM used insted of a consequent ENCODER_PULSES_PER_STEP? If you want to make one menu item per step use `#define ENCODER_STEPS_PER_MENU_ITEM 1`.